src/main/java/io/github/oliviercailloux/y2018/apartments/apartment/ApartmentFactory.java

```diff
@@ -154,39 +152,53 @@ private static double simulateRandomDraw(double mean, double deviation) {
 	/**
 	 * Call an API which generates a random address. This function aims at getting
 	 * the random address generated.
-	 *
-	 * @param retry indicates how many time the function has been called (Retry
-	 *              System). It will allow us to try escaping problems while
-	 *              contacting API (if it fails once, we try to call it again until
-	 *              the number specified in the class attribute NB_MAX_RETRY).
+	 * <p>
+	 * So, we generate a random latitude and a longitude and try to
+	 * {@link io.github.oliviercailloux.y2018.apartments.utils.JsonConvert#getAddressFromJson(String)
+	 * retrieve an address}
+	 * <p>
+	 * 
 	 * @return the address generated.
-	 * @throws IOException if we cannot contact the API generator.
+	 * @throws InvalidObjectException in case the API doesn't return a good format
+	 *                                after a certain number of attempts (RETRY)
 	 */
-	private static String getRandomAddress(int retry) throws IOException {
-
-		String address = "";
-		// Code from
-		// https://www.developpez.net/forums/d1354479/java/general-java/recuperer-reponse-d-adresse-http/
-		try (InputStream is = new URL(URL_API_ADDRESS).openConnection().getInputStream()) {
-			BufferedReader reader = new BufferedReader(new InputStreamReader(is, "utf-8"));
-			StringBuilder builder = new StringBuilder();
-			for (String line = reader.readLine(); line != null; line = reader.readLine()) {
-				builder.append(line + "\n");
+	private static String getOnlineRandomAddress() throws InvalidObjectException {
+		/**
+		 * Maximum test value in case the API does not return the
+		 * features.properties.labels field In the case of an API error (for example
+		 * 500, or others), jax-rs throws an error of type
+		 * <code>ClientErrorException</code>
+		 */
+		final int RETRY = 5;
```

Comment:
Comme déjà discuté, je ne trouve pas cette façon de réessayer élégante, et je ne vois pas l’utilité de réessayer de toute façon. À supprimer.

Reply:
Cette fois si cela concerne un cas plus grand. En effet, le serveur peut nous fournir une informations incomplète dans le cas où la longitude et la latitude ne corresponde pas à une adresse. Dans ce cas, le champs des données est vide. Ce cas est très régulier et nous avons jugé ce retry nécessaire. Il ne concerne pas le fait que l'API soit momentanément indisponible mais le fait que les paramètres que nous lui envoyons ne correspondent pas à une maison.
Souhaitez vous tout de même la supprimer ? 
Dans ce cas, il y a de très nombreuses chance pour que le programme plante lors de son exécution dans le scénario nominal actuel.

Reply:
Lors de la génération de 500 adresses par API nous avons vu que l'API retourné parfois des champs vides (aux alentours de 50 fois sur 500 je dirais)
Mais, généralement au bout de 2 essais, nous récupérons une adresse.
Nous avons mis ici 5 essais mais il s'agit d'un cas de figure qui semble hors de portée : ce n'est qu'une simple prévention

Reply:
It was an error